src/analyze/analyze-security.c

```diff
@@ -119,9 +122,22 @@ struct security_assessor {
         bool default_dependencies_only;
 };
 
-static void security_info_free(struct security_info *i) {
+static SecurityInfo *security_info_new(void) {
+        SecurityInfo *info = new(SecurityInfo, 1);
```

Comment:
OOM check is still required

Reply:
those checks are done after declaration. So for example on line 2222 of the same file. returning log_oom here would create problems right because the function returns type SecurityInfo *.

Reply:
if `new` fails here, it will crash when `info` is dereferenced immediately below. The usual pattern in these cases is not to return log_oom from here, but simply return NULL, and treat that as sign of oom from the caller

Reply:
and also keep the log_ooms after declaration?